distributed/dashboard/components/scheduler.py

```diff
@@ -0,0 +1,1797 @@
+            text_font_size=value("10pt"),
+        )
+        self.root.text(
+            source=self.source,
+            text="done",
+            y="bottom",
+            x="right",
+            x_offset=-5,
+            text_align="right",
+            text_font_size=value("10pt"),
+        )
+        self.root.ygrid.visible = False
+        self.root.yaxis.minor_tick_line_alpha = 0
+        self.root.yaxis.visible = False
+        self.root.xgrid.visible = False
+        self.root.xaxis.minor_tick_line_alpha = 0
+        self.root.xaxis.visible = False
+
+        hover = HoverTool(
+            point_policy="follow_mouse",
+            tooltips="""
+                <div>
+                    <span style="font-size: 14px; font-weight: bold;">Name:</span>&nbsp;
+                    <span style="font-size: 10px; font-family: Monaco, monospace;">@name</span>
+                </div>
+                <div>
+                    <span style="font-size: 14px; font-weight: bold;">All:</span>&nbsp;
+                    <span style="font-size: 10px; font-family: Monaco, monospace;">@all</span>
+                </div>
+                <div>
+                    <span style="font-size: 14px; font-weight: bold;">Memory:</span>&nbsp;
+                    <span style="font-size: 10px; font-family: Monaco, monospace;">@memory</span>
+                </div>
+                <div>
+                    <span style="font-size: 14px; font-weight: bold;">Erred:</span>&nbsp;
+                    <span style="font-size: 10px; font-family: Monaco, monospace;">@erred</span>
+                </div>
+                <div>
+                    <span style="font-size: 14px; font-weight: bold;">Ready:</span>&nbsp;
+                    <span style="font-size: 10px; font-family: Monaco, monospace;">@processing</span>
+                </div>
+                """,
+        )
+        self.root.add_tools(hover)
+
+    @without_property_validation
+    def update(self):
+        with log_errors():
+            state = {"all": valmap(len, self.plugin.all), "nbytes": self.plugin.nbytes}
+            for k in ["memory", "erred", "released", "processing", "waiting"]:
+                state[k] = valmap(len, self.plugin.state[k])
+            if not state["all"] and not len(self.source.data["all"]):
+                return
+
+            d = progress_quads(state)
+
+            update(self.source, d)
+
+            totals = {
+                k: sum(state[k].values())
+                for k in ["all", "memory", "erred", "released", "waiting"]
+            }
+            totals["processing"] = totals["all"] - sum(
+                v for k, v in totals.items() if k != "all"
+            )
+
+            self.root.title.text = (
+                "Progress -- total: %(all)s, "
+                "in-memory: %(memory)s, processing: %(processing)s, "
+                "waiting: %(waiting)s, "
+                "erred: %(erred)s" % totals
+            )
+
+
+class MemoryUse(DashboardComponent):
```

Comment:
I *think* that we no longer use this?  Maybe we should though (or something like it)

Reply:
Yeah it's not used, but it looks useful. I've created a new individual plot for it for now.